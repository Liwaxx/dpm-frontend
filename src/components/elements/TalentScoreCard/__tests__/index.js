import React from 'react';
import renderer from 'react-test-renderer';
import Component from '../index';

jest.mock('../../../elements/Text');

describe('TalentScoreCard', () => {
  it('renders correctly', () => {
    const data = {
      name: 'UI',
      rank: [{ name: 'Achie', previous: 2, point: 5 }, { name: 'Muhammad', previous: 5, point: 4.8 }, { name: 'Charlie', previous: 1, point: 4.6 }]
    };
    const tree = renderer
      .create(<Component data={data} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with custom class', () => {
    const tree = renderer
      .create(<Component data={{}} className="test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});