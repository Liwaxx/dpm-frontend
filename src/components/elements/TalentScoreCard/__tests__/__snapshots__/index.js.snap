// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalentScoreCard renders correctly 1`] = `
<div
  className="Component-root-9"
>
  <table
    className="Component-content-3"
  >
    <thead>
      <tr>
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
      </tr>
      <tr>
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
      </tr>
      <tr>
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
        <td
          className="Component-rowCell-10"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TalentScoreCard renders with custom class 1`] = `
<div
  className="Component-root-9 test"
>
  <table
    className="Component-content-3"
  >
    <thead>
      <tr>
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
        <th
          className="Component-headerCell-4"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
